explorer/src/components/account/TokenHistoryCard.tsx

```diff
@@ -148,7 +171,20 @@ function TokenHistoryTable({ tokens }: { tokens: TokenInfoWithPubkey[] }) {
       return oldestSlot !== undefined && tx.slot >= oldestSlot;
     });
 
-  if (mintAndTxs.length === 0) {
+  const filtered = mintAndTxs.filter(({ mint }) => {
+    if (filter === "") {
+      return true;
+    }
+    return mint.toBase58() === filter;
+  });
+
+  React.useEffect(() => {
+    if (!fetching && filtered.length < 1 && !allFoundOldest) {
```

Comment:
I don't think `allFoundOldest` is appropriate here. Seems like it should be something like "filteredFoundOldest"